h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoReader.java

```diff
@@ -38,6 +31,44 @@ protected void readModelData() throws IOException {
 
   @Override
   protected XGBoostMojoModel makeModel(String[] columns, String[][] domains, String responseColumn) {
-    return new XGBoostMojoModel(columns, domains, responseColumn);
+    byte[] boosterBytes;
+    try {
+      boosterBytes = readblob("boosterBytes");
+    } catch (IOException e) {
+      throw new IllegalStateException("MOJO is corrupted: cannot read the serialized Booster", e);
+    }
+    if (useJavaScoring()) {
+      return makeJavaModel(columns, domains, responseColumn, boosterBytes);
+    } else {
+      return makeNativeModel(columns, domains, responseColumn, boosterBytes);
+    }
+  }
+
+  public static boolean useJavaScoring() {
+    return Boolean.getBoolean("sys.ai.h2o.xgboost.scoring.java.enable");
   }
+
+  private  XGBoostMojoModel makeNativeModel(String[] columns, String[][] domains, String responseColumn,
+                                            byte[] boosterBytes) {
+    XGBoostNativeMojoModel model = new XGBoostNativeMojoModel(columns, domains, responseColumn);
+    InputStream is = new ByteArrayInputStream(boosterBytes);
```

Comment:
Should the IS be closed automatically ? I'd use a try-with-resources block here.

Reply:
In general yes but this is ByteArrayInputStream, close() doesn't do anything - it is empty.